### PR TITLE
Fix RGB Color Temp Swaps

### LINKED ISSUE #449: detect_non_ha_changes incorrectly detecting changes from rgb_color to color_temp
#### Version information:
version 1.4.1

#### Description:
I have 2 govee lightstrips in the Kitchen controlled by Adaptive Lighting, both are members of an HA group called "Kitchen Lightstrips" which is what I set AL to control. 

AL works correctly and adjusts the color/brightness but doing so causes them to get flagged as manual control, when no manual control actions were done. I'm not sure how to avoid this, but this problem isn't happening with my other lights/rooms in the house with seperate AL configurations. 

See below for logs, these 2 the lines appear odd to me, but i'm not sure how to avoid this. 
```
'light.kitchen_lightstrip_bot' **switched from RGB mode to color_temp or visa versa**
'adaptive_lighting.**manual_control**' event fired for switch.adaptive_lighting_kitchen for light light.kitchen_lightstrip_bot
```

Logs
```
2023-03-15 09:37:42.733 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] Detected a 'light.kitchen_lightstrip_bot' 'state_changed' event: '{'min_color_temp_kelvin': 2000, 'max_color_temp_kelvin': 9009, 'min_mireds': 111, 'max_mireds': 500, 'supported_color_modes': [<ColorMode.COLOR_TEMP: 'color_temp'>, <ColorMode.HS: 'hs'>], 'color_mode': <ColorMode.HS: 'hs'>, 'brightness': 255, 'hs_color': (26.739, 36.078), 'rgb_color': (255, 204, 163), 'xy_color': (0.424, 0.366), 'rate_limit_total': 100, 'rate_limit_remaining': 51, 'rate_limit_reset_seconds': -1678887462.71, 'rate_limit_reset': '1969-12-31T19:00:00', 'rate_limit_on': 5, 'manufacturer': 'Govee', 'model': 'H6159', 'friendly_name': 'Kitchen Lightstrip Bot', 'supported_features': 19}' with context.id='01GVJQV9SMZWY9074JB60P6VE6'
…
2023-03-15 09:37:52.639 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] Detected a 'light.kitchen_lightstrip_bot' 'state_changed' event: '{'min_color_temp_kelvin': 2000, 'max_color_temp_kelvin': 9009, 'min_mireds': 111, 'max_mireds': 500, 'supported_color_modes': [<ColorMode.COLOR_TEMP: 'color_temp'>, <ColorMode.HS: 'hs'>], 'color_mode': <ColorMode.HS: 'hs'>, 'brightness': 255, 'hs_color': (26.739, 36.078), 'rgb_color': (255, 204, 163), 'xy_color': (0.424, 0.366), 'rate_limit_total': 100, 'rate_limit_remaining': 51, 'rate_limit_reset_seconds': -1678887472.62, 'rate_limit_reset': '1969-12-31T19:00:00', 'rate_limit_on': 5, 'manufacturer': 'Govee', 'model': 'H6159', 'friendly_name': 'Kitchen Lightstrip Bot', 'supported_features': 19}' with context.id='01GVJQVKF8K1YSFRWN2BAKRETJ'
…
2023-03-15 09:38:22.527 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] Kitchen: '_update_attrs_and_maybe_adapt_lights' called with context.id='adapt_lgt:b'^pQZ':interval:b'1O''
2023-03-15 09:38:22.529 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] Adaptive Lighting: Kitchen: '_adapt_lights(['light.kitchen_lightstrip_top', 'light.kitchen_lightstrip_bot'], 45.0, force=False, context.id=adapt_lgt:b'^pQZ':interval:b'1O')' called
2023-03-15 09:38:22.530 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] Kitchen: Setting color_temp of light light.kitchen_lightstrip_top
2023-03-15 09:38:22.531 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] State of 'light.kitchen_lightstrip_top' didn't change wrt change event nr. 0 (context.id=adapt_lgt:b'^pQZ':interval:b'1O')
2023-03-15 09:38:22.532 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] Kitchen: Scheduling 'light.turn_on' with the following 'service_data': {'entity_id': 'light.kitchen_lightstrip_top', 'brightness': 255, 'color_temp_kelvin': 3995} with context.id='adapt_lgt:b'^pQZ':interval:b'1O''
2023-03-15 09:38:22.535 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] Kitchen: Setting color_temp of light light.kitchen_lightstrip_bot
2023-03-15 09:38:22.536 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] color RGB of 'light.kitchen_lightstrip_bot' significantly changed from (255, 136, 12) to (255, 204, 163) with context.id='adapt_lgt:b'^pQZ':interval:b'1O''
2023-03-15 09:38:22.536 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] color RGB of 'light.kitchen_lightstrip_bot' significantly changed from (255, 136, 12) to (255, 204, 163) with context.id='adapt_lgt:b'^pQZ':interval:b'1O''
2023-03-15 09:38:22.536 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] 'light.kitchen_lightstrip_bot' switched from RGB mode to color_temp or visa versa
2023-03-15 09:38:22.536 DEBUG (MainThread) [custom_components.adaptive_lighting.switch] 'adaptive_lighting.manual_control' event fired for switch.adaptive_lighting_kitchen for light light.kitchen_lightstrip_bot
```

### PR DESCRIPTION
**Pull request by th3w1zard1**
If any attribute is missing during `_attributes_have_changed` convert the ones that exist into the ones the other has.
Fixes #449

@basnijholt there's two branches with the name 'fix_rgb_color_temp'. The one that is not marked 2 can be deleted. I don't have perms 😄

### CODE REVIEW

File: custom_components/adaptive_lighting/switch.py [hunk 1]
```diff
@@ -643,6 +644,94 @@ def color_difference_redmean(
     return math.sqrt(red_term + green_term + blue_term)
 
 
+def color_xy_to_cct(
+    x: float,
+    y: float,
+) -> float:
+    n = (x - 0.3320) / (0.1858 - y)
```
Comment (basnijholt) [inline custom_components/adaptive_lighting/switch.py:651]: Could you add a reference perhaps of where these magic numbers come from 😄

File: custom_components/adaptive_lighting/switch.py [hunk 2]
```diff
@@ -643,6 +644,94 @@ def color_difference_redmean(
...
+    return 437 * (n**3) + 3601 * (n**2) + 6861 * n + 5517
+
+
+def color_rgb_to_cct(
+    red: float,
+    green: float,
+    blue: float,
+) -> float:
+    n = ((0.23881) * red + (0.25499) * green + (-0.58291) * blue) / (
```
Comment (basnijholt) [inline custom_components/adaptive_lighting/switch.py:660]: here too

**Comment by basnijholt** [comment]
I pushed some changes, let me know whether you like to revert or not?

**Comment by th3w1zard1** [comment]
e.g. If both `new_attributes` and `old_attributes` both have `ATTR_COLOR_TEMP_KELVIN` there's no need to convert.

**Comment by basnijholt** [comment]
These calculations are likely extremely cheap.

However, I believe this commit should fix it https://github.com/basnijholt/adaptive-lighting/pull/514/commits/3a26135fcfa79d25b31d53e0c547b862bf1917f0.

**Review by basnijholt — APPROVED** [review]

**Comment by basnijholt** [comment]
Go ahead with squash merging if you are happy with it :)

**Comment by th3w1zard1** [comment]
> Go ahead with squash merging if you are happy with it :)

Doesn't pass the test?

**Comment by basnijholt** [comment]
Sorry about that! I’ll fix that ASAP (tomorrow) 😅

**Comment by basnijholt** [comment]
> e.g. If both new_attributes and old_attributes both have ATTR_COLOR_TEMP_KELVIN there's no need to convert. What's the performance impact of converting anyway?

I actually struggle to see that this happens, perhaps I am missing the point?

```python
    if not any(
        attr in old_attributes and attr in new_attributes
        for attr in [ATTR_COLOR_TEMP_KELVIN, ATTR_RGB_COLOR]
    ):
        old_attributes = _convert_attributes(old_attributes)
        new_attributes = _convert_attributes(new_attributes)
```

Have we only call `_convert_attributes` if old and new both have `ATTR_COLOR_TEMP_KELVIN` then this would evaluate like:
```python
    if not any(
        ATTR_COLOR_TEMP_KELVIN in old_attributes and ATTR_COLOR_TEMP_KELVIN in new_attributes
    ): ...
```

```python
    if not any(
        True
    ): ...
```

```python
    if False: ...
```

So it wouldn't get executed/

**Comment by th3w1zard1** [comment]
@basnijholt Ah yes, you're right! This looks good and it still passes the test.

**Pull request merged**